### DISCUSSION TITLE
Confused by Client.transport type not being Transport

### DISCUSSION THREAD

**Discussion by frangio** [wevm/viem#778] [General]
I expected `client.transport` to be what I understand as a transport, i.e. something I can pass into a client constructor, of type `Transport`. But I found that it isn't, and I learned that transports are functions and that `client.transport` is (more or less) the return value of that function.

I'd think the `transport` property of clients should have a different name. I would also expect to be able to recover the actual (?) transport out of a client.

The documentation could mention this distinction between the transport function and its result as well.

**Comment by tmm** [comment]
Good point! Naming definitely could be improved here. (No concrete proposal on my end right now. Just commenting so you know this has been seen.)
